src/NSubstitute/NSubstitute.csproj

```diff
@@ -53,4 +53,12 @@
     <DefineConstants>$(DefineConstants);SYSTEM_DIAGNOSTICS_CODEANALYSIS_NULLABILITY</DefineConstants>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(CI)' == 'True'">
+    <!--Deterministic Build -->
+    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+  </PropertyGroup>
+  <ItemGroup>
+     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All"/>
+  </ItemGroup>
 </Project>
```

Comment:
**question:** sorry for the silly questions but i'm unfamiliar with sourcelink.

* should we only reference the package for CI builds or when `/p:ContinuousIntegrationBuild=true`?
* if we moved to another source provider, how do builds of commits with this package work?

Reply:
There is no problem to include this package on all the builds as it's a private asset. 

Also for other source providers there are other packages. See https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink